Add region parameter on spot and spots_by_query

zydeco opened the pull request:
This adds the optional region parameter, supported on the Text Search and Details requests:
* https://developers.google.com/places/web-service/search
* https://developers.google.com/places/web-service/details

region — The region code, specified as a [ccTLD](https://en.wikipedia.org/wiki/Country_code_top-level_domain) (country code top-level domain) two-character value. Most ccTLD codes are identical to ISO 3166-1 codes, with some exceptions. This parameter will only influence, not fully restrict, search results. If more relevant results exist outside of the specified region, they may be included. When this parameter is used, the country name is omitted from the resulting formatted_address for results in the specified region.

edwinwills commented:
Hi @zydeco - thanks so much for contributing, this looks like a handy addition.

Do you mind adding a test for this option, and either updating the existing VCR cassettes, or ensuring that an empty `region` parameter does not get sent to Google's API (this is why the new change broke the tests - it's an unrecognised HTTP request, as it has the new `region` parameter in it, even though it's empty).

zydeco commented:
Oops, I hadn't checked the tests and thought empty parameters were not sent.
I've fixed it now: the empty region will not be sent, and I added a tests for requests using the region parameter.

edwinwills commented on spec/google_places/spot_spec.rb:
```diff
@@ -140,6 +140,20 @@
+      end
+    end
+
+    it 'should be a collection of Spots' do
```
`should not include country in formatted address`

edwinwills commented:
Nice one - thanks! 👍
